romancal/dark_current/dark_current_step.py

```diff
@@ -53,11 +54,28 @@ def process(self, input):
             if "nresultants" not in dark_model.meta.exposure:
                 dark_model.meta.exposure["nresultants"] = dark_model.data.shape[0]
 
+            # get the dark slope and dark slope error from the reference file & trim ref pixels
+            dark_slope = dark_model.dark_slope[4:-4, 4:-4]
+            dark_slope_err = dark_model.dark_slope_error[4:-4, 4:-4]
+
+            # get the resultant info from the input file
+            read_pattern = input_model.meta.exposure.read_pattern
+            frame_time = input_model.meta.exposure.frame_time
+
+            # get the average time for each resultant
+            time_resultants = (
+                np.array([np.average(x) for x in read_pattern], dtype=np.float32)
+                * frame_time
+            )
+            # get the time for the dark file
+            time_dark = np.sum(time_resultants)
+
             # Do the dark correction
             out_model = input_model
-            nresultants = len(input_model.meta.exposure["read_pattern"])
-            out_model.data -= dark_model.data[:nresultants]
-            out_model.pixeldq |= dark_model.dq
+            out_model.data -= dark_slope * time_dark
```

Comment:
I think this can be just out_model.data -= dark_slope?  Since the dark subtraction now follows the ramp fitting, data is in units of DN / s, and dark_slope is also in DN / s, so we shouldn't need to multiply by the time?